### PR TITLE
added multi-word responses

### PR DESCRIPTION
**Pull request by PatrickAlphaC**
This adds a really simple overview of the multi-word response and large responses. 

I've put a note at the top that it's currently limited in support. Happy to make that bigger if needed.

### CODE REVIEW

**Comment by PatrickAlphaC** [comment]
It's much more thorough after a force push update now!

File: docs/Example Job Specs/multi-word-job.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
+---
+layout: nodes.liquid
+date: Last Modified
+title: 'MultiWord Example Job Spec'
+permalink: "docs/example-job-spec-multi-word/"
+---
+
+This is an example v1 job spec for returning multiple responses in 1 Chainlink API Call. 
+
+```JSON
```
Comment (StephenFluin) [inline docs/Example Job Specs/multi-word-job.md:10]: Can you make sure we're have TOML versions?

File: docs/Oracle Jobs/adapters.md [hunk 2]
```diff
@@ -422,4 +422,96 @@ req.addUint("until", now + 1 hours);
     }
   ]
 }
-```
\ No newline at end of file
+```
+[block:api-header]
```
Comment (StephenFluin) [inline docs/Oracle Jobs/adapters.md:426]: Can you use markdown headers instead? `##`

File: docs/Using Any API/large-responses.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
+---
+layout: nodes.liquid
+section: smartContract
+date: Last Modified
+title: "Large Responses"
+permalink: "docs/large-responses/"
+whatsnext: {"Make an Existing Job Request":"/docs/existing-job-request/", "API Reference":"/docs/chainlink-framework/", "Contract Addresses":"/docs/decentralized-oracles-ethereum-mainnet/"}
+metadata: 
+  title: "Large Responses"
```
Comment (StephenFluin) [inline docs/Using Any API/large-responses.md:9]: I don't think we use this title anywhere?

File: docs/Using Any API/make-a-http-get-request.md [hunk 4]
```diff
@@ -4,7 +4,7 @@ section: smartContract
 date: Last Modified
 title: "Make a GET Request"
 permalink: "docs/make-a-http-get-request/"
-whatsnext: {"Make an Existing Job Request":"/docs/existing-job-request/", "API Reference":"/docs/chainlink-framework/", "Contract Addresses":"/docs/decentralized-oracles-ethereum-mainnet/"}
+whatsnext: {"Make an Existing Job Request":"/docs/existing-job-request/", "API Reference":"/docs/chainlink-framework/", "Contract Addresses":"/docs/decentralized-oracles-ethereum-mainnet/", "Multi-Variable and Large Responses":"/docs/multi-variable-responses/"}
```
Comment (StephenFluin) [inline docs/Using Any API/make-a-http-get-request.md:7]: I think you split this into two pages so the title doesn't match?

File: docs/Using Any API/multi-variable-responses.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
+---
+layout: nodes.liquid
+section: smartContract
+date: Last Modified
+title: "Multi-Variable Responses"
+permalink: "docs/multi-variable-responses/"
+whatsnext: {"Make an Existing Job Request":"/docs/existing-job-request/", "API Reference":"/docs/chainlink-framework/", "Contract Addresses":"/docs/decentralized-oracles-ethereum-mainnet/", "Large Responses": "/docs/large-responses/"}
+metadata: 
+  title: "Multi-Variable Responses"
```
Comment (StephenFluin) [inline docs/Using Any API/multi-variable-responses.md:9]: I don't think we need

**Review by StephenFluin — CHANGES_REQUESTED** [review]

File: docs/Using Any API/large-responses.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    emit RequestFulfilled(requestId, bytesData);
+    data = bytesData;
+    image_url = string(data);
+  }
+
+  ////---- END ----////
+}
+```
+
+> The job spec for the Chainlink node in this example can be [found here](../example-job-spec-large/).
```
Comment (dmitrydao) [inline docs/Using Any API/large-responses.md:104]: Please use the following format for callouts:
> 📘 Job Spec
> 
> The job spec for the Chainlink node in this example can be [found here](../example-job-spec-large/).

We're currently using [Readme styling](https://rdmd.readme.io/docs/callouts) so one-line callouts are not highlighted in the docs. Will fix that later.

File: docs/Using Any API/large-responses.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+The `oracle` keyword refers to a specific Chainlink node that a contract makes an API call from, and the `specId` refers to a specific job for that node to run. Each job is unique and returns different types of data. 
+
+For example, a job that returns a `bytes32` variable from an API would have a different `specId` than a job that retrieved the same data, but in the form of a `uint256` variable.
+
+[market.link](https://market.link/) provides a searchable catalogue of Oracles, Jobs and their subsequent return types.
+
+# Choosing an Oracle Job without specifying the URL
+
+If your contract is calling a public API endpoint, an Oracle job may already exist for it. If so, it could mean you do not need to add the URL, or other adapter parameters into the request, since the job already configured to return the desired data. This makes your smart contract code more succinct. To see an example of a contract using an existing job which calls the <a href="https://www.coingecko.com/en/api#explore-api" target="_blank">CoinGecko API</a>, see [Make an Existing Job Request](../existing-job-request/).
```
Comment (dmitrydao) [inline docs/Using Any API/large-responses.md:118]: Replace `<a>` with a markdown version.

File: docs/Using Any API/multi-variable-responses.md [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+> Making a GET request will fail unless your deployed contract has enough LINK to pay for it. **Learn how to [Acquire testnet LINK](../acquire-link/) and [Fund your contract](../fund-your-contract/)**.
+> 
+
+<div class="remix-callout">
+    <a href="https://remix.ethereum.org/#optimize=false&evmVersion=null&gist=dcf554e03223427aa4c7aef099d98429&runs=200&version=soljson-v0.8.6+commit.11564f7e.js" target="_blank" class="cl-button--ghost solidity-tracked">Deploy a Multi-Word Contract Example in Remix ↗</a>
+    <a href="../deploy-your-first-contract/" title="">What is Remix?</a>
+</div>
+
+```javascript Kovan
+pragma solidity ^0.8.0;
```
Comment (dmitrydao) [inline docs/Using Any API/multi-variable-responses.md:35]: The code styling for solidity works so let's replace javascript with solidity

File: docs/Using Any API/multi-variable-responses.md [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+  title: "Multi-Variable Responses"
+  description: "Learn how to make a GET request to an API and receive multiple responses from a smart contract, using Chainlink."
+  image: 
+    0: "/files/930cbb7-link.png"
+---
+This page explains how to make an HTTP GET request to an external API from a smart contract, using Chainlink's [Request & Receive Data](../request-and-receive-data/) cycle and then receive multiple responses.
+
+This is known as **multi-variable** or **multi-word** responses. 
+
+>⚠️ NOTE: This feature is available as of Chainlink `0.10.10` but can only be customized with different URLs and responses on the node operator side. In `0.10.11` generic jobs will be available for solidity as well. Please see [Make an API Call](../make-a-http-get-request/) for more information on making API calls.
```
Comment (dmitrydao) [inline docs/Using Any API/multi-variable-responses.md:18]: Let's use the following styling by now:
> ⚠️ NOTE
> 
> This feature is available as of Chainlink `0.10.10` but can only be customized with different URLs and responses on the node operator side. In `0.10.11` generic jobs will be available for solidity as well. Please see [Make an API Call](../make-a-http-get-request/) for more information on making API calls.

File: docs/Using Any API/large-responses.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+  title: "Large Responses"
+  description: "Learn how to make a GET request to an API and receive a dynamically sized large response from a smart contract, using Chainlink.."
+  image: 
+    0: "/files/930cbb7-link.png"
+---
+This page explains how to make an HTTP GET request to an external API from a smart contract, using Chainlink's [Request & Receive Data](../request-and-receive-data/) cycle and then receive large responses.
+
+# Large Response
+
+To consume an API with a large responses, your contract should inherit from <a href="https://github.com/smartcontractkit/chainlink/blob/master/contracts/src/v0.8/ChainlinkClient.sol" target="_blank">`ChainlinkClient`</a>. This contract exposes a struct called `Chainlink.Request`, which your contract should use to build the API request. The request should include the oracle address, the job id, the fee, adapter parameters, and the callback function signature.
```
Comment (dmitrydao) [inline docs/Using Any API/large-responses.md:18]: Replace `<a>` with a markdown version

File: docs/Using Any API/large-responses.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+> Making a GET request will fail unless your deployed contract has enough LINK to pay for it. **Learn how to [Acquire testnet LINK](../acquire-link/) and [Fund your contract](../fund-your-contract/)**.
+> 
+
+<div class="remix-callout">
+    <a href="https://remix.ethereum.org/#optimize=false&evmVersion=null&gist=10ae2499f48f2932b38f34a96b124443&runs=200&version=soljson-v0.8.6+commit.11564f7e.js" target="_blank" class="cl-button--ghost solidity-tracked">Deploy this contract using Remix ↗</a>
+    <a href="../deploy-your-first-contract/" title="">What is Remix?</a>
+</div>
+
+```javascript Kovan
+//SPDX-License-Identifier: MIT
```
Comment (dmitrydao) [inline docs/Using Any API/large-responses.md:31]: Replace javascript with solidity

File: docs/Using Any API/multi-variable-responses.md [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+    emit RequestMultipleFulfilled(requestId, usdResponse, eurResponse, jpyResponse);
+    usd = usdResponse;
+    eur = eurResponse;
+    jpy = jpyResponse;
+  }
+  ////---- END ----////
+}
+```
+
+> The job spec for the Chainlink node in this example can be [found here](../example-job-spec-multi-word/).
```
Comment (dmitrydao) [inline docs/Using Any API/multi-variable-responses.md:112]: Use the same callouts styling as I mentioned on another page

File: docs/Using Any API/multi-variable-responses.md [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+
+The `oracle` keyword refers to a specific Chainlink node that a contract makes an API call from, and the `specId` refers to a specific job for that node to run. Each job is unique and returns different types of data. 
+
+For example, a job that returns a `bytes32` variable from an API would have a different `specId` than a job that retrieved the same data, but in the form of a `uint256` variable.
+
+[market.link](https://market.link/) provides a searchable catalogue of Oracles, Jobs and their subsequent return types.
+
+# Choosing an Oracle Job without specifying the URL
+
+If your contract is calling a public API endpoint, an Oracle job may already exist for it. If so, it could mean you do not need to add the URL, or other adapter parameters into the request, since the job already configured to return the desired data. This makes your smart contract code more succinct. To see an example of a contract using an existing job which calls the <a href="https://www.coingecko.com/en/api#explore-api" target="_blank">CoinGecko API</a>, see [Make an Existing Job Request](../existing-job-request/).
```
Comment (dmitrydao) [inline docs/Using Any API/multi-variable-responses.md:126]: Replace `<a>` with a markdown version

File: docs/Using Any API/multi-variable-responses.md [hunk 14]
```diff
@@ -0,0 +1,128 @@
...
+---
+This page explains how to make an HTTP GET request to an external API from a smart contract, using Chainlink's [Request & Receive Data](../request-and-receive-data/) cycle and then receive multiple responses.
+
+This is known as **multi-variable** or **multi-word** responses. 
+
+>⚠️ NOTE: This feature is available as of Chainlink `0.10.10` but can only be customized with different URLs and responses on the node operator side. In `0.10.11` generic jobs will be available for solidity as well. Please see [Make an API Call](../make-a-http-get-request/) for more information on making API calls.  
+
+# MultiWord
+
+To consume an API with multiple responses, your contract should inherit from <a href="https://github.com/smartcontractkit/chainlink/blob/master/contracts/src/v0.8/ChainlinkClient.sol" target="_blank">`ChainlinkClient`</a>. This contract exposes a struct called `Chainlink.Request`, which your contract should use to build the API request. The request should include the oracle address, the job id, the fee, adapter parameters, and the callback function signature.
```
Comment (dmitrydao) [inline docs/Using Any API/multi-variable-responses.md:22]: Replace `<a>` with a markdown version

File: docs/Using Any API/large-responses.md [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+  )
+    public
+    recordChainlinkFulfillment(requestId)
+  {
+    emit RequestFulfilled(requestId, bytesData);
+    data = bytesData;
+    image_url = string(data);
+  }
+
+  ////---- END ----////
```
Comment (dmitrydao) [inline docs/Using Any API/large-responses.md:100]: I haven't seen us using this in any of our code samples. Let's remove it.

File: docs/Using Any API/multi-variable-responses.md [hunk 16]
```diff
@@ -0,0 +1,128 @@
...
+  )
+    public
+    recordChainlinkFulfillment(requestId)
+  {
+    emit RequestMultipleFulfilled(requestId, usdResponse, eurResponse, jpyResponse);
+    usd = usdResponse;
+    eur = eurResponse;
+    jpy = jpyResponse;
+  }
+  ////---- END ----////
```
Comment (dmitrydao) [inline docs/Using Any API/multi-variable-responses.md:108]: I haven't seen us using this in any of our code samples. Let's remove it.

File: docs/Using Any API/large-responses.md [hunk 17]
```diff
@@ -0,0 +1,120 @@
...
+
+```javascript Kovan
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@chainlink/contracts/src/v0.8/ChainlinkClient.sol";
+
+/**
+ * @notice DO NOT USE THIS CODE IN PRODUCTION. This is an example contract. 
+ */
```
Comment (dmitrydao) [inline docs/Using Any API/large-responses.md:38]: Maybe add a note after the code sample on how would people use it in production?

File: docs/Using Any API/large-responses.md [hunk 18]
```diff
@@ -0,0 +1,120 @@
...
+   * Oracle: 0xc57B33452b4F7BB189bB5AfaE9cc4aBa1f7a4FD8 (Chainlink DevRel)
+   *
+   */
+  constructor(
+  ) {
+    setChainlinkToken(0xa36085F69e2889c224210F603D836748e7dC0088);
+    setChainlinkOracle(0xc57B33452b4F7BB189bB5AfaE9cc4aBa1f7a4FD8);
+  }
+
+  ////---- VARIABLE BYTES ARRAY RETURNED BY ORACLE ----////
```
Comment (dmitrydao) [inline docs/Using Any API/large-responses.md:62]: We mostly use the following comments style for section or function descriptions:
```
    /**
    * Comment text
    */
```

For smaller comments, I've seen `// comment text` variation but I haven't seen `////----` one.
Should we replace it?

**Review by dmitrydao — CHANGES_REQUESTED** [review]

File: docs/Example Job Specs/multi-word-job.md (hunk 1 above)
Comment (PatrickAlphaC) [reply docs/Example Job Specs/multi-word-job.md:10]: Not yet :(

0.10.11 will have the TOML that we need and then I can update.

File: docs/Using Any API/large-responses.md (hunk 3 above)
Comment (PatrickAlphaC) [reply docs/Using Any API/large-responses.md:9]: We use it in `navigation.js`

File: docs/Using Any API/multi-variable-responses.md (hunk 5 above)
Comment (PatrickAlphaC) [reply docs/Using Any API/multi-variable-responses.md:9]: You don't think we need this page?

File: docs/Using Any API/large-responses.md (hunk 6 above)
Comment (PatrickAlphaC) [reply docs/Using Any API/large-responses.md:104]: Ahh ok. thanks!

File: docs/Using Any API/large-responses.md (hunk 17 above)
Comment (PatrickAlphaC) [reply docs/Using Any API/large-responses.md:38]: What do you mean?

Comment (dmitrydao) [reply docs/Using Any API/large-responses.md:38]: I mean if we state that people shouldn't use this in the production, then the logical question - what they should use instead, maybe some caveats or whatever else they have to consider when using in the production.

File: docs/Oracle Jobs/adapters.md [hunk 19]
```diff
@@ -9,12 +9,12 @@ Core adapters are the built-in functionality that every Chainlink node supports.
...
-| Signed Integers   | [EthInt256](#ethint256)  | int256             |
+| Name              | Core Adapter              | Ethereum Data Type |
+| ----------------- | ------------------------- | ------------------ |
+| Signed Integers   | [EthInt256](#ethint256)   | int256             |
 | Unsigned Integers | [EthUint256](#ethuint256) | uint256            |
 | Bytes             | [EthBytes32](#ethbytes32) | bytes32            |
-| Boolean           | [EthBool](#ethbool)    | bool               |
+| Boolean           | [EthBool](#ethbool)       | bool               |
 
 You can learn more about Solidity data types <a href="https://docs.soliditylang.org/en/v0.5.3/types.html" target="_blank">here</a>.
```
Comment (dmitrydao) [inline docs/Oracle Jobs/adapters.md:19]: Replace <a> with a markdown version

File: docs/Using Any API/large-responses.md [hunk 20]
```diff
@@ -0,0 +1,118 @@
...
+  title: "Large Responses"
+  description: "Learn how to make a GET request to an API and receive a dynamically sized large response from a smart contract, using Chainlink.."
+  image: 
+    0: "/files/930cbb7-link.png"
+---
+This page explains how to make an HTTP GET request to an external API from a smart contract, using Chainlink's [Request & Receive Data](../request-and-receive-data/) cycle and then receive large responses.
+
+# Large Response
+
+To consume an API with a large responses, your contract should inherit from <a href="https://github.com/smartcontractkit/chainlink/blob/master/contracts/src/v0.8/ChainlinkClient.sol" target="_blank">`ChainlinkClient`</a>. This contract exposes a struct called `Chainlink.Request`, which your contract should use to build the API request. The request should include the oracle address, the job id, the fee, adapter parameters, and the callback function signature.
```
Comment (dmitrydao) [inline docs/Using Any API/large-responses.md:18]: Replace <a> with a markdown version

File: docs/Using Any API/make-a-http-get-request.md [hunk 21]
```diff
@@ -149,4 +149,4 @@ The data type returned by a specific job depends on the [adapters](../core-adapt
 
 If your contract is calling a public API endpoint, an Oracle job may already exist for it. If so, it could mean you do not need to add the URL, or other adapter parameters into the request, since the job already configured to return the desired data. This makes your smart contract code more succinct. To see an example of a contract using an existing job which calls the <a href="https://www.coingecko.com/en/api#explore-api" target="_blank">CoinGecko API</a>, see [Make an Existing Job Request](../existing-job-request/).
```
Comment (dmitrydao) [inline docs/Using Any API/make-a-http-get-request.md:150]: Replace <a> with a markdown version

File: docs/Using Any API/multi-variable-responses.md (hunk 12 above)
Comment (dmitrydao) [reply docs/Using Any API/multi-variable-responses.md:112]: > 📘 Note
> 
> The job spec for the Chainlink node in this example can be [found here](../example-job-spec-multi-word/).

**Review by dmitrydao — DISMISSED** [review]

File: docs/Example Job Specs/large-response-job.md [hunk 22]
```diff
@@ -0,0 +1,49 @@
...
+layout: nodes.liquid
+date: Last Modified
+title: 'MultiWord Example Job Spec'
+permalink: "docs/example-job-spec-large/"
+---
+
+This is an example v1 job spec for returning multiple responses in 1 Chainlink API Call. 
+
+```JSON
+{
```
Comment (dmitrydao) [inline docs/Example Job Specs/large-response-job.md:11]: Wait, don't we need to use the TOML version of the specs from now on?

File: docs/Oracle Jobs/adapters.md (hunk 19 above)
Comment (PatrickAlphaC) [reply docs/Oracle Jobs/adapters.md:19]: This PR isn't to clean up old code. Can we focus on the changes I'm making?

File: docs/Using Any API/large-responses.md (hunk 17 above)
Comment (PatrickAlphaC) [reply docs/Using Any API/large-responses.md:38]: It's saying it's just an example contract, and not to use that specific contract in prod. I'll clarify.

File: docs/Using Any API/make-a-http-get-request.md (hunk 21 above)
Comment (PatrickAlphaC) [reply docs/Using Any API/make-a-http-get-request.md:150]: Let's focus on what this PR is doing.

File: docs/Example Job Specs/large-response-job.md (hunk 22 above)
Comment (PatrickAlphaC) [reply docs/Example Job Specs/large-response-job.md:11]: No we do, the TOML version of this job spec isn't ready. We will have to convert it to TOML laster

File: docs/Using Any API/multi-variable-responses.md [hunk 23]
```diff
@@ -0,0 +1,130 @@
...
+    usd = usdResponse;
+    eur = eurResponse;
+    jpy = jpyResponse;
+  }
+}
+```
+
+> 📘 Note
+> 
+> The job spec for the Chainlink node in this example can be [found here](../example-job-spec-multi-word/). .
```
Comment (StephenFluin) [inline docs/Using Any API/multi-variable-responses.md:114]: One tiny typo. Can you get rid of the extra `. ` and rebase?

**Review by StephenFluin — CHANGES_REQUESTED** [review]
One tiny typo, then I think everything is good to go!

**Comment by PatrickAlphaC** [comment]
The rebasing sort of sucks since I have to force push everytime :/

**Comment by StephenFluin** [comment]
@PatrickAlphaC Force pushing is the right thing to do, because you are literally rewriting your branch's history to remove intermediate states. If you could remove intermediate states without force pushing, that would be super dangerous.

**Review by StephenFluin — APPROVED** [review]

**Comment by PatrickAlphaC** [comment]
@dmitrydao let me know when we are good.

**Pull request merged**